class/practice7.1/MaybeTheBestCodeEver.cpp

```diff
@@ -0,0 +1,56 @@
+#include <iostream>
+#include <print>
+#include "Rectangle.hpp"
+
+int main() {
+    double widths[5] = {0.0};
+    double heights[5] = {0.0};
+    int count = 0;
+    double biggestArea = 0;
+    double smallestArea = 1000000;
+
+    for (int i = 0; i < 5; ++i) {
+        std::println("Enter rectangle {}: ", i + 1);
+        if (std::cin >> widths[i] >> heights[i]) {
+            Rectangle rect(widths[i], heights[i]);
+            double area = rect.area();
+            count++;
+            if (area > biggestArea) {
+                biggestArea = area;
+            }
+            if (area < smallestArea) {
+                smallestArea = area;
+            }
+        }
+    }
+
+    for (int i = 0; i < 5; ++i) {
+        for (int j = 0; j < 5; ++j) {
+            if (i != j) {
+                Rectangle r1(widths[i], heights[i]);
```

Comment:
Here we recreating Rectangle objects which may be a big overhead. Let's just store them in container